app/controllers/builds_controller.rb

```diff
@@ -8,36 +8,21 @@ class BuildsController < ApplicationController
 
   def index
     @builds = current_project.builds.order('id desc').page(params[:page])
-
-    respond_to do |format|
-      format.html
-      format.json { render json: @builds }
-    end
   end
 
   def new
     @build = current_project.builds.build
   end
 
   def create
-    @build = create_build
+    @build = new_or_modified_build
     @build.creator = current_user
-    @build.save
-
-    start_docker_build if @build.persisted? && params[:build_image].present?
```

Comment:
also fixing that previously failed updates where ignored since they were 'persisted?'